src/libraries/System.Private.CoreLib/src/System/Numerics/TotalOrderIeee754Comparer.cs

```diff
@@ -0,0 +1,206 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System.Collections.Generic;
+using System.Diagnostics.CodeAnalysis;
+
+namespace System.Numerics
+{
+    /// <summary>
+    /// Represents a comparison operation that compares floating-point numbers
+    /// with IEEE 754 totalOrder semantic.
+    /// </summary>
+    /// <typeparam name="T">The type of the numbers to be compared, must be an IEEE 754 floating-point type.</typeparam>
+    public class TotalOrderIeee754Comparer<T> : IComparer<T>, IEqualityComparer<T>
+        where T : IFloatingPointIeee754<T>?
```

Comment:
We've typically overridden object.Equals/GetHashCode on our comparers.  Should we be doing the same here? e.g.
```C#
public override bool Equals(object? obj) =>
    obj is not null && GetType() == obj.GetType();
```
(the above assumes the type is sealed; otherwise, it would need to be more like `obj is TotalOrderIeee754Comparer<T>`)

Reply:
Analyzer forces `IEquatable<T>` when overriding `Equals`, which is not in the approved API.

Reply:
@stephentoub, what's your preference here?

Should we ensure `Equals/GetHashCode` is overridden and disable the analyzer for this file? Implement `IEquatable` as well, or just leave it as is?

Reply:
@huoyaoyuan, if you could handle the above then we can get this merged.